.github/workflows/test.yml

```diff
@@ -39,68 +36,107 @@ jobs:
       fail-fast: false
       matrix:
         include:
+          # Test against old and new python, and traefik v2 and v3
           - python: "3.8"
-            backend: "etcd"
-          - python: "3.9"
-            backend: "consul"
-          - python: "3.10"
-            backend: "redis"
-          - python: "3.11"
-    steps:
-      # NOTE: In GitHub workflows, environment variables are set by writing
-      #       assignment statements to a file. They will be set in the following
-      #       steps as if would used `export MY_ENV=my-value`.
-      - name: Configure environment variables
-        run: |
-          echo "PATH=$PWD/bin:$PATH" >> $GITHUB_ENV
+            backend: file
+            traefik-version: "v2.10.7"
+          - python: "3.12"
+            backend: file
+            traefik-version: "v3.0.0-beta5"
 
+          # Test each backend that requires a python client against a modern
+          # version of python to ensure clients are compatible.
+          #
+          # - About redis versions:
+          #   redis can be installed from source, as an ubuntu distribution, or
+          #   snap package. To avoid complexity we don't build from source, but we
+          #   test against the ubuntu and snap package, where the ubuntu package
+          #   typically is behind the snap package.
+          #
+          #   redis-test-version is provided as a matrix input only to provide
+          #   visibility via github's web UI.
+          #
+          - python: "3.12"
+            backend: redis
+            install-via: apt # version 6.0.16 in ubuntu 22.04, https://packages.ubuntu.com/jammy/redis-server
+            redis-test-version: "6.0.16?"
```

Comment:
not something you need to spend time on, but I'd omit the redundant `-test-` in the version names (`redis-version`, etc. which matches e.g. `traefik-version`). `-test-` is doesn't add any information.

Reply:
Resolved by [3c5d754](https://github.com/jupyterhub/traefik-proxy/pull/231/commits/3c5d754da7bdc4c5037688efcc8dad577cb16af0)